monkey/monkey_island/cc/ui/src/components/configuration-components/ReformatHook.js

```diff
@@ -65,6 +65,7 @@ export function formatCredentialsForForm(credentialsData = []) {
     for (const existingRow of existingRows) {
       let canMutate = true;
 
+      if(!secret) break;
```

Comment:
The function is called `formatCredentialsForForm` but it looks like it's merging rows that have same ID's but different types of secrets. I would rather we had this logic separated.